
instructions for creating version with hashes as before - known to be working. 
eval "$(/rds/project/rds-5mCMIDBOkPU/ma595/fenics/conda/new-conda/miniconda3/bin/conda shell.bash hook)"
conda create --name static-versions
conda activate static-versions
conda install conda-build anaconda-client
conda config --set anaconda_upload no
conda-build -c conda-forge ./ --skip-existing --debug |& tee -a conda-build.out

conda search 'xtensor[channel=conda-forge, subdir=linux-64]'
conda search 'petsc[channel=conda-forge, subdir=linux-64]'
conda search conda-forge::petsc
conda search 'petsc=3.16.0'
conda search 'petsc=3.16.0 *real*'


fenics-os:
~/fenics/original-recipe/newrecipe/recipe

# error 1
CMake Error at cpp/CMakeLists.txt:43 (find_package):
  Could not find a configuration file for package "xtensor" that is
    compatible with requested version range "0.23.10...<0.25".

# error 2 
-- Found Python3: /home/ubuntu/fenics/miniconda3/envs/florianorig/bin/python3.8 (found version "3.8.13") found components: Interpreter Development.Module
CMake Error at python/CMakeLists.txt:39 (find_package):
  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.

git show -s --format=%ci f960aa1fd0184e82f34d1cd5bf7a7f581089fd01

# error 3
  File "/home/ubuntu/fenics/miniconda3/envs/florianorig/lib/python3.8/site-packages/conda/models/records.py", line 345, in combined_depends
    result = {ms.name: ms for ms in MatchSpec.merge(self.depends)}
  File "/home/ubuntu/fenics/miniconda3/envs/florianorig/lib/python3.8/site-packages/conda/models/match_spec.py", line 478, in merge
    reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
  File "/home/ubuntu/fenics/miniconda3/envs/florianorig/lib/python3.8/site-packages/conda/models/match_spec.py", line 478, in <lambda>
    reduce(lambda x, y: x._merge(y, union), group) if len(group) > 1 else group[0]
  File "/home/ubuntu/fenics/miniconda3/envs/florianorig/lib/python3.8/site-packages/conda/models/match_spec.py", line 509, in _merge
    final = this_component.merge(that_component)
  File "/home/ubuntu/fenics/miniconda3/envs/florianorig/lib/python3.8/site-packages/conda/models/match_spec.py", line 760, in merge
    raise ValueError("Incompatible component merge:\n  - %r\n  - %r"
ValueError: Incompatible component merge:
  - 'mpi_openmpi_*'
  - 'mpi_openmpi_tempest*'
